website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,92 @@
+
+#### Integrating with Operator Lifecycle Manager
+
+The Operator Lifecycle Manager (OLM) is often used to distribute operators via operator catalogs. The operator deployment object in an OLM integrated operator is defined in the `ClusterServiceVersion` yaml. It is important to remember to set the node affinity block in the each of the `spec.template.spec` under `spec.install.deployments`.
+
+### Examples
+This section attempts to show how each of the sample operators could be updated to declare a set of supported architectures. Most of the time, these changes will be in `yaml` files, but different kinds of operators embed `PodSpec` and `PodTemplateSpec` objects in different places.
+
+#### Go Operators
+The most common directory to look at for objects that will need an architecture affinity block will is the operators `config` directory. In the go samples app, located in the `testdata directory`, you'll find several Deployment configsthat need this updated including:
+`testdata/go/v2/memcached-operator/config/default/manager_auth_proxy_patch.yaml`
+`testdata/go/v2/memcached-operator/config/default/manager_webhook_patch.yaml`
+`testdata/go/v2/memcached-operator/config/manager/manager.yaml`
+ 
+ Occasionally, go operators also define objects dynamically. An example of this kind of object would be in `testdata/go/v2/memcached-operator/controllers/memcached_controller.go`:
+ 
+ ```go
+Template: corev1.PodTemplateSpec{
+    ObjectMeta: metav1.ObjectMeta{
+		Labels: ls,
+	},
+	Spec: corev1.PodSpec{
+		Containers: []corev1.Container{{
+			Image:   "memcached:1.4.36-alpine",
+			Name:    "memcached",
+			Command: []string{"memcached", "-m=64", "-o", "modern", "-v"},
+			Ports: []corev1.ContainerPort{{
+			    ContainerPort: 11211,
+				Name:          "memcached",
+			}},
+		}},
+	},
+},
+ ```
+
+Here, you would update the PodSpec by referencing the API directly, rather than supplying a yaml. The final result might look like this:
+ ```go
+	Spec: corev1.PodSpec{
+        Affinity:
+            NodeAffinity:
+                RequiredDuringSchedulingIgnoredDuringExecution:
+                    NodeSelectorTerms: []corev1.NodeSelectorTerm {{
+                       MatchExpressions: []corev1.NodeSelectorRequirement {{
+                           key: "kubernetes.io/arch"
+                           operator: "In"
+                           values: []string{"amd64", "arm64", "ppc64le", "s390x"}
+                       }} 
+                    }}
+	Spec: corev1.PodSpec{
+		Containers: []corev1.Container{{
+			Image:   "memcached:1.4.36-alpine",
+			Name:    "memcached",
+			Command: []string{"memcached", "-m=64", "-o", "modern", "-v"},
+			Ports: []corev1.ContainerPort{{
+			    ContainerPort: 11211,
+				Name:          "memcached",
+			}},
+		}},
+	},
+ ```
```

Comment:
this example is not accurate it would be like:


```go
	Spec: corev1.PodSpec{
					// TODO(user): If you project provide support to other platforms
					// then you must to change the NodeAffinity accordingly.
					// More info: https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#affinity-and-anti-affinity
					Affinity: &corev1.Affinity{
						NodeAffinity: &corev1.NodeAffinity{
							RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
								NodeSelectorTerms: []corev1.NodeSelectorTerm{
									{
										MatchExpressions: []corev1.NodeSelectorRequirement{
											{
												Key:      "kubernetes.io/arch",
												Operator: "In",
												Values:   []string{"amd64"},
											},
											{
												Key:      "kubernetes.io/os",
												Operator: "In",
												Values:   []string{"linux"},
											},
										},
									},
								},
							},
						},
					},
					SecurityContext: &corev1.PodSecurityContext{
```

Reply:
Thank you for providing this! :+1: 
I'm new to golang syntax, especially that which involves these API spec references. This was very helpful.